ui/DesignSystem/Component/Transaction.svelte

```diff
@@ -17,25 +19,43 @@
 
   export let transaction = null;
   export let payer = null;
-  export let subject = null;
+
+  const subject = formatSubject(transaction.messages[0]);
+  const subjectAvatarShape = () => {
+    switch (subject.type) {
+      case SubjectType.User:
+      case SubjectType.Member:
+      case SubjectType.UserProject:
+        return "circle";
+      default:
+        return "square";
+    }
+  };
 </script>
 
 <Caption style="color: var(--color-foreground-level-6); margin-bottom: 16px">
   Your transaction
 </Caption>
+
 <Row dataCy="summary" variant={transaction.id ? 'top' : 'single'}>
-  <div slot="left">
+  <div slot="left" data-cy="message">
     <Title>{formatMessage(transaction.messages[0])}</Title>
   </div>
 
-  <div slot="right">
-    <Avatar
-      dataCy="subject-avatar"
-      title={subject.name}
-      imageUrl={subject.imageUrl}
-      avatarFallback={subject.avatarFallback}
-      variant={subject.variant === Variant.User ? 'circle' : 'square'}
-      style="color: var(--color-foreground)" />
+  <div slot="right" data-cy="subject">
+    {#await subject.avatarSource then avatar}
```

Comment:
Don't think it's ideal that components in the design system are dependent on outcomes of side-effects. It'd be better if the result of this is passed as a prop.

Reply:
das macht sinn